src/App.tsx

```diff
@@ -1,23 +1,92 @@
 import "./App.css";
-
+import React, { useState } from 'react';
 import { taskList } from "./taskList";
 
+let taskListCopy = taskList;
+
+// Create a list with additional field to mark deleted task
+  let taskListExt: any[] = [];
+  for (let i=0 ; i < taskListCopy.length; i++){
+    taskListExt.push(
+      {
+        'id'  : taskListCopy[i].id,
+        'name': taskListCopy[i].name,
+        'del' : '' 
+      }
+    )
+  }
+
+
+//
+
 function TaskShow() {
+  let [tasks, setTasks ] = useState(taskListExt);
+  let [newTask, setNewTask] = useState('');
+  
+  const handleSubmit:React.FormEventHandler<HTMLFormElement> = (event) => {
+    
+    event.preventDefault();
+    let newId  = tasks[ tasks.length-1 ].id + 1;
+    
+      setTasks( (previousTask) => [
+                  ...previousTask,
+                  { id: newId,
+                    name: newTask
+                  }
+              ]
+            )
+  }
+
   return (
+
     <div className="App">
+
       <h2>My Fancy TODO App</h2>
-      {taskList.map((task) => (
-        <div className="container">
-            <div className="id-container">
+      {tasks.map((task) => (
+        <div className="container" key={task.id}>
+            <div className="id-container" >
               <span className="text">{task.id}</span>
             </div>
             <div className="task-name-container">
               <span className="text">{task.name}</span>
             </div>
+
+            <div className="delete-button">
+              <button type='button'
+                onClick={ () => {
+                  task.del = '*'
+                  setTasks(tasks.filter( item => item.del !== '*'))
+                }}  
+              >
+               <font size="+3" color='red'>🗑</font>
+   
+              </button>
+            </div>  
         </div>
       ))}
+
+      <div className="new-task-container"> 
+        <form onSubmit={handleSubmit} >
+          <div>
+            <label>Task name
+              <div className="newtask-name">
+                <input 
+                    style={{width: "640px"}}
+                    type='text'
+                    placeholder="New task name here"
+                    onChange={ event => setNewTask(event.target.value)}
+                    value={newTask}
+                />
+                <button type='submit' className="newtask-button">
+                    Add new
+                </button>
+              </div>  
+            </label>     
```

Comment:
Hierarchically, this should not be structured this way where the `label` contains the submit `button`
The reason is that a `form` can have multiple `input`s (and a `label` for each of them) while there is only 1 submit `button`

Instead, I would suggest structuring it similar to this
```
<form>
  <div className="task-name-input">
    <label for="task-name">Task name</label>
    <input name="task-name"/>
  </div>

  <button type="submit">Add new</button>
</form>
```